pkg/controller/certificatemanager/certificatemanager.go

```diff
@@ -185,6 +193,8 @@ func (cm *certificateManager) GetOrCreateKeyPair(cli client.Client, secretName,
 			log.V(3).Info("secret %s has invalid DNS names, the expected names are: %v", secretName, dnsNames)
 			return keyPair, nil
 		}
+	} else if keyPair == nil {
+		log.V(1).Info("Keypair wasn't found, create a new one", "namespace", secretNamespace, "name", secretName)
```

Comment:
I believe there is a case where the keypair is found, but it's not valid or out of date, and so we still fall through to creating a new keypair - this log isn't the same as "We need a new keypair". rather, it's specifically "We need a new keypair because we  couldn't find one on the cluster".